src/super_gradients/module_interfaces/supports_input_shape_check.py

```diff
@@ -0,0 +1,60 @@
+from typing import Tuple, Optional
+
+
+class SupportsInputShapeCheck:
+    def validate_input_shape(self, input_shape: Tuple[int, ...]) -> None:
+        """
+        Validate the input shape of the model.
+        If the validation passes, the method should return None.
+        If the validation fails, the method should raise a ValueError.
+
+        :param input_shape: (Tuple[int, ...]) Input shape (usually BCHW)
+        :return: None.
+        """
+        exact_shape = self.get_exact_input_shape_size()
+        if exact_shape is not None:
+            # Take n last elements of input shape, where n is the number of elements in exact_shape
+            shape_to_check = input_shape[-len(exact_shape) :]
+            if any(shape_to_check != exact_shape):
+                raise ValueError(f"Input shape {input_shape} is not supported. Model requires input with spatial size of {exact_shape}.")
+
+        min_shape = self.get_minimum_input_shape_size()
+        if min_shape is not None:
+            shape_to_check = input_shape[-len(min_shape) :]
+            if any(shape_to_check < min_shape):
+                raise ValueError(f"Input shape {input_shape} is not supported. Model requires input with spatial size of at least {min_shape}.")
+
+        shape_steps = self.get_input_shape_steps()
+        if shape_steps is not None:
+            shape_to_check = input_shape[-len(shape_steps) :]
+            if any(shape_to_check % shape_steps != 0):
+                raise ValueError(f"Input shape {input_shape} is not supported. Model requires input with spatial size that is a multiple of {shape_steps}.")
+
+    def get_exact_input_shape_size(self) -> Optional[Tuple[int, ...]]:
+        """
+        Get the exact input shape of the model.
+        Class should override this method if the model requires an exact input shape.
+        Usually true for transformer-based models.
+        :return:
+        """
+        return None
+
+    def get_minimum_input_shape_size(self) -> Optional[Tuple[int, ...]]:
+        """
+        Get the minimum input shape of the model.
+        Class should override this method if the model requires a minimum input shape.
+        Usually true for transformer-based models.
+        :return:
+        """
+        return None
+
+    def get_input_shape_steps(self) -> Optional[Tuple[int, ...]]:
```

Comment:
`shape_steps` in `get_input_shape_steps` is not super intuitive name IMO.
I called it `shape_multiple` in the `AutoPadding`, I think it's a little clearer.
I also thought of `shape_modulo` but I think its a bit less clear as well. What do you think?